tests/data/test1485

```diff
@@ -0,0 +1,51 @@
+<testcase>
```

Comment:
Should we also have test cases for HTTP/2 (and maybe HTTP/3) ?

Reply:
There is no 101 protocol switch for HTTP/3, though. For HTTP/2 this could be done.

Reply:
Hmm, but this is about having the `CURLINFO` things available by the time header callback is called for the (last) empty header though, and not 101 protocol switch.

Reply:
Yes, the PR fixes the issue you raised, I believe...😌

Reply:
Right, so the other protocol go through the same codepaths in the end.